core/src/com/unciv/ui/worldscreen/bottombar/BattleTable.kt

```diff
@@ -236,11 +236,20 @@ class BattleTable(val worldScreen: WorldScreen): Table() {
         attackerNameWrapper.add(UnitGroup(attacker.unit,25f)).padRight(5f)
         attackerNameWrapper.add(attackerLabel)
         add(attackerNameWrapper)
-
+        var canNuke = true
         val defenderNameWrapper = Table()
         for (tile in targetTile.getTilesInDistance(Battle.NUKE_RADIUS)) {
             val defender = tryGetDefenderAtTile(tile, true)
             if (defender == null) continue
+
+            //To make sure we dont nuke civilisations we cant declare war with
+            val attackerciv = attacker.getCivInfo()
+            val defenderciv = tile.getCity()?.civInfo
+            if(defenderciv != null && defenderciv.diplomacy[attackerciv.civName]!= null) {
+                val canDeclareWar = attackerciv.getDiplomacyManager(defenderciv).canDeclareWar()
```

Comment:
Another problem is that `canDeclareWar()` returns `false` in case you are already in the war state.
Thus, you will not be able to nuke your real enemy! That's a problem...

Reply:
oh u are right! I totally missed that lol. just fixed it!